hls-plugin-api/src/Ide/Types.hs

```diff
@@ -1084,11 +1086,11 @@ mkFormattingHandlers f = mkPluginHandler SMethod_TextDocumentFormatting ( provid
         mf <- lift $ getVirtualFile $ toNormalizedUri uri
         case mf of
           Just vf -> do
-            let typ = case m of
-                  SMethod_TextDocumentFormatting -> FormatText
-                  SMethod_TextDocumentRangeFormatting -> FormatRange (params ^. L.range)
+            let (typ, mtoken) = case m of
```

Comment:
Seems strange to have the same code in both branches if it doesn't depend on the scrutinee `case m of` ..
I'd add separate `mtoken = params ^. L.workDoneToken` in the let.

Reply:
It actually _does_ depend on the scrutinee, because `params` has type `MethodParams m`, so we don't know what it is until we do the case. In this two cases we actually have different types for `params`, they just happen to boht have `range` fields. A bit weird, but you have to do it like this!

Reply:
Ah, these invisible type-level things don't make reviewing easy :smile:

Reply:
haha yeah